plugins/badges/src/api/BadgesClient.ts

```diff
@@ -18,46 +18,126 @@ import { generatePath } from 'react-router-dom';
 import { ResponseError } from '@backstage/errors';
 import { Entity, DEFAULT_NAMESPACE } from '@backstage/catalog-model';
 import { BadgesApi, BadgeSpec } from './types';
-import { DiscoveryApi, IdentityApi } from '@backstage/core-plugin-api';
+import {
+  ConfigApi,
+  DiscoveryApi,
+  IdentityApi,
+} from '@backstage/core-plugin-api';
 
 export class BadgesClient implements BadgesApi {
   private readonly discoveryApi: DiscoveryApi;
   private readonly identityApi: IdentityApi;
+  private readonly configApi: ConfigApi;
 
   constructor(options: {
     discoveryApi: DiscoveryApi;
     identityApi: IdentityApi;
+    configApi: ConfigApi;
   }) {
     this.discoveryApi = options.discoveryApi;
     this.identityApi = options.identityApi;
+    this.configApi = options.configApi;
+  }
+
+  static fromConfig(options: {
+    discoveryApi: DiscoveryApi;
+    identityApi: IdentityApi;
```

Comment:
We can avoid having to use the IdentityApi by instead using the `FetchApi` which does this automatically. It also takes care of discovery for you so you can fetch `plugin://badges/some/interesting/path`
https://backstage.io/docs/reference/core-plugin-api.fetchapiref